core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1125,6 +1127,11 @@ class KafkaApis(val requestChannel: RequestChannel,
     )
 
     val responseTopics = authorizedRequestInfo.map { topic =>
+      val isClientRequest = offsetRequest.replicaId == ListOffsetsRequest.CONSUMER_REPLICA_ID
+      if (isClientRequest) {  // Brokers send listOffset requests too to check if truncation needed --- ignore those.
+        listOffsetsRequestInstrumentation.logUsage(request.context.principal, topic)
```

Comment:
Maybe also add info about why this is needed (by ng) in the comments.

Reply:
Details are linked in the ticket (in commit message).

I personally think we should not tangle NG information in the Kafka code.  What do you think.

Reply:
I think it doesn't matter really, as the commit message has the same visibility with code. btw is this repo readable by outside linkedin?

Reply:
This is, but NG is not.

At least we hide NG under the (unreachable) ticket link.

Reply:
Oh re-read your comment.  Now I agree with you.  Let me add it.

Reply:
Added